### PR TITLE
EVSS Migration - 526 Submit - Fixes

### PR DESCRIPTION
**Pull request by sethdarragile6**
## Summary

- The PR contains fixes that arose while testing against the latest submit526 endpoint changes in Lighthouse sandbox. It contains some temporary workarounds for some known Lighthouse issues (identified with comments in the code view)
- Disability Benefits Experience Team 1 (DBEX-TREX 🦖)
- Changes are currently gated behind the `disability_compensation_lighthouse_submit_migration` flipper switch.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#69080
- [link](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/69080)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb [hunk 1]
```diff
@@ -164,6 +163,13 @@ def transform_service_pay(service_pay_source)
 
       private
 
+      def populate_te(lh_request_body)
```
Comment (sethdarragile6) [inline lib/evss/disability_compensation_form/form526_to_lighthouse_transform.rb:166]: as noted in comments above, this is a temporary workaround for a current Lighthouse bug that was blocking testing

File: lib/lighthouse/benefits_claims/service.rb [hunk 2]
```diff
@@ -97,14 +97,16 @@ def submit526(body, lighthouse_client_id = nil, lighthouse_rsa_key_path = nil, o
...
-      end
+
+      body = {
+        data: {
+          type: 'form/526',
+          attributes: body
+        }
+      }.as_json.deep_transform_keys { |k| k.camelize(:lower) }
+
+      # TODO: determine why camelcasing is malfunctioning for this field and revisit the below
```
Comment (christinec-fftc) [inline lib/lighthouse/benefits_claims/service.rb:108]: mystery answered? perhaps add a comment below as to why the override is needed?

Comment (sethdarragile6) [reply lib/lighthouse/benefits_claims/service.rb]: Good call. Updated!

**Review by aurora-a-k-a-lightning — DISMISSED** [review]

File: lib/lighthouse/benefits_claims/service.rb [hunk 3]
```diff
@@ -119,6 +122,16 @@ def submit526(body, lighthouse_client_id = nil, lighthouse_rsa_key_path = nil, o
 
     private
 
+    def fix_current_va_employee(body)
+      if body.dig('data', 'attributes', 'veteranIdentification')&.select do |field|
+           field['currentVAEmployee']
```
Comment (tommasina-va) [inline lib/lighthouse/benefits_claims/service.rb:127]: Perfect use case for `dig`!

File: lib/lighthouse/benefits_claims/service.rb [hunk 4]
```diff
@@ -119,6 +122,16 @@ def submit526(body, lighthouse_client_id = nil, lighthouse_rsa_key_path = nil, o
...
+         end&.key?('currentVAEmployee')
+        body['data']['attributes']['veteranIdentification']['currentVaEmployee'] =
```
Comment (tommasina-va) [inline lib/lighthouse/benefits_claims/service.rb:129]: Not sure I'm following this line. Is it necessary?

**Review by tommasina-va — DISMISSED** [review]
Nice work! 👏🏻

**Comment by rmtolmach** [comment]
Error from failed checks:
```
Fetching gem metadata from https://rubygems.org/.......
Git error: command `git reset --hard 8233129def00c29b33e56ac5ad6d08f5442d665c`
in directory
/home/runner/work/vets-api/vets-api/vendor/bundle/ruby/3.2.0/bundler/gems/betamocks-8233129def00
has failed.
fatal: Could not parse object '8233129def00c29b33e56ac5ad6d08f5442d665c'.
Error: The process '/opt/hostedtoolcache/Ruby/3.2.2/x64/bin/bundle' failed with exit code 11
```
I see it on another PR as well. I'm looking into it since I'm on Backend support today.

**Comment by rmtolmach** [comment]
I noticed it failed after Fetching gem metadata from https://rubygems.org/....... so I checked here: https://status.rubygems.org/ and it's reporting "Service Under Maintenance" 
Try re-running it a bit later, I guess.

**Review by rmtolmach — APPROVED** [review]

**Comment by sethdarragile6** [comment]
> I noticed it failed after Fetching gem metadata from https://rubygems.org/....... so I checked here: https://status.rubygems.org/ and it's reporting "Service Under Maintenance" Try re-running it a bit later, I guess.

Thank you. I neglected to mention that I was all green- until I updated the branch with master. I'll try another update in a couple hours

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by tommasina-va — APPROVED** [review]

**Comment by sethdarragile6** [comment]
> I noticed it failed after Fetching gem metadata from https://rubygems.org/....... so I checked here: https://status.rubygems.org/ and it's reporting "Service Under Maintenance" Try re-running it a bit later, I guess.

@rmtolmach I just updated my branch with master and I'm still failing on that linting check for some reason. Are you still looking into it?

**Comment by rmtolmach** [comment]
@sethdarragile6 hmm... No, I wasn't still looking into it because I thought it was caused by something related to rubygems. And it also started working in the other PR it was failing in :/

**Review by NB28VT — APPROVED** [review]
Just some non-blocking suggestions on comments in the code.

Other thing I might note is it seems like a lot of the metadata in these test cases has reversed ("yes" to "no", "new" to "increase" stuff like that), so that would be an uninformed concern, but if you've been working through that in detail trust you know what you did here!

**Comment by sethdarragile6** [comment]
> Just some non-blocking suggestions on comments in the code.
> 
> Other thing I might note is it seems like a lot of the metadata in these test cases has reversed ("yes" to "no", "new" to "increase" stuff like that), so that would be an uninformed concern, but if you've been working through that in detail trust you know what you did here!

Thanks, @NB28VT ! Those little tweaks were made in order to pass Lighthouse API validation, and better reflect actual expected data states. I think that when the "with_everything" JSON was set up, the intent was simply to exercise as much of the possible object graph as possible and didn't care whether it made "business sense".

**Pull request merged**